Cache

pschupp01 opened the pull request.

pschupp01 committed:
Add Cache and Cache test able to fetch and store

pschupp01 committed:
add invalidate method on Cache and test it

pschupp01 committed:
Fixed tests on cache, using mocha instead of Jest. Fetch, cache, invalidate and invalidate(id) are currently tested

rricard commented on package.json:
```diff
@@ -9,7 +9,7 @@
     "data-develop": "nodemon data/index.js",
     "develop": "npm run data-develop & npm run ui-start",
     "data-test": "mocha --compilers js:babel/register --recursive -r co-mocha data/tests",
-    "ui-test": "jest",
```
let jest here

rricard commented on package.json:
```diff
@@ -9,7 +9,7 @@
     "data-develop": "nodemon data/index.js",
     "develop": "npm run data-develop & npm run ui-start",
     "data-test": "mocha --compilers js:babel/register --recursive -r co-mocha data/tests",
```
mocha --compilers js:babel/register --recursive -r co-mocha data/tests src/lib/tests

rricard commented on src/lib/tests/Cache.js:
```diff
@@ -0,0 +1,74 @@
+  describe('cacheFetch', function() {
+    it('should fetch the url properly for the first time', function(done) {
+      this.cache.fetch("ID","URL1")
+      .then(res=> {
```
``` js
this.cache.fetch("ID","URL1")
.then(res => {
  assert.equal(res,"URL1");
  return true;
})
.then(() => done(), done);
```

or 

``` js
it('should fetch the url properly for the first time', function*() {
  const res = yield this.cache.fetch("ID","URL1");
  assert.equal(res,"URL1");
});
```

pschupp01 committed:
cache tests now use generators instead of done function to throw errors on assertion

pschupp01 committed:
Cache works now with native Android/IOs storage using AsyncStorage, it is enable on texts documents
Cache tests are completely commented as the reactnative AsyncStorage is not mocked yet

rricard commented on src/components/FileViewer/TextViewer.js:
```diff
@@ -5,6 +5,8 @@ import React, {
   Text
 } from "react-native";
 
+import Cache from "./../../lib/Cache.js";
```
``` js
import {
  fetchWithCache,
  invalidateCache
} from "./../../lib/Cache.js";
```

rricard commented on src/lib/Cache.js:
```diff
@@ -0,0 +1,30 @@
+  AsyncStorage
+} from "react-native";
+
+export default function Cache() {};
```
Non-ES6 form

``` js
export default class Cache {}
```

But we won't do that anyway

rricard commented on src/lib/Cache.js:
```diff
@@ -0,0 +1,30 @@
+} from "react-native";
+
+export default function Cache() {};
+Cache.fetch=(url)=> {
```
``` js
export function fetchWithCache( ...) {
  ...
}
```

rricard commented on src/lib/Cache.js:
```diff
@@ -0,0 +1,30 @@
+  })
+};
+
+Cache.invalidate=(id)=>{
```
``` js
export function invalidateCache( ...) {
  ...
}
```

rricard commented on src/lib/tests/Cache.js:
```diff
@@ -0,0 +1,49 @@
+// import assert from "assert";
```
Delete instead of commenting

pschupp01 committed:
Cache.js now export the function needed to fetchwithCache and invalidate cache

pschupp01 committed:
Add an attribute lastModified on Files to know when was the last file's update

pschupp01 committed:
Cache now check the lastModified date to know if a file in cache is out of date and should be replace

pschupp01 committed:
Cache.js is now typed with Flow

rricard commented:
Ok, well, that's good!
